helpers/env-variables.ts

```diff
@@ -362,9 +362,37 @@ const getToolEnvs = (tools?: Tool[]): EnvVar[] => {
   return toolEnvs;
 };
 
-const getSystemPromptEnv = (tools?: Tool[]): EnvVar => {
-  const defaultSystemPrompt =
-    "You are a helpful assistant who helps users with their questions.";
+const getSystemPromptEnv = (
+  tools?: Tool[],
+  dataSources?: TemplateDataSource[],
+): EnvVar => {
+  let defaultSystemPrompt = "";
+  if (dataSources?.find((ds) => ds.type === "file" || ds.type === "web")) {
+    defaultSystemPrompt = `'You\\'re a helpful assistant! You have provided information from a knowledge base that has been passed to you in nodes of information.
+Each node has useful metadata such as node ID, file name, page, etc.
+Please add the citation to the data node for each sentence or paragraph that you reference in the provided information.
+The citation format is: . [(<citation_order_number>)][<citation_order_number>]
+At the end of the answer, please provide the citation details in the following format:
+[<citation_order_number>]: <node_id> "<node_id> or <file_name> or <url>"
```

Comment:
```suggestion
[<citation_order_number>]: <node_id> "<node_id> or <file_name> or <url>"
```
why would we want file_name or URL?